tests/common/posix/aws_test_posix_utils.c

```diff
@@ -255,3 +252,85 @@ TEST( Full_POSIX_UTILS, UTILS_strnlen )
 }
 
 /*-----------------------------------------------------------*/
+
+TEST( Full_POSIX_UTILS,  UTILS_TimespecCompare )
+{
+    struct timespec xTimeX = { .tv_sec = 0, .tv_nsec = 0 };
+    struct timespec xTimeY = { .tv_sec = 0, .tv_nsec = 0 };
+
+    /* Check return value with NULL parameters. */
+    TEST_ASSERT_EQUAL_INT( 0, UTILS_TimespecCompare( NULL, NULL) );
+    TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecCompare( NULL, &xTimeY ) );
+    TEST_ASSERT_EQUAL_INT( 1, UTILS_TimespecCompare( &xTimeX, NULL ) );
+
+    /* check equal value */
+    TEST_ASSERT_EQUAL_INT( 0, UTILS_TimespecCompare( &xTimeX, &xTimeY) );
+
+    /* check greater value */
+    xTimeX.tv_nsec = 1;
+    TEST_ASSERT_EQUAL_INT( 1, UTILS_TimespecCompare( &xTimeX, &xTimeY ) );
+    xTimeX.tv_sec = 1;
+    TEST_ASSERT_EQUAL_INT( 1, UTILS_TimespecCompare( &xTimeX, &xTimeY ) );
+
+    /* check less than value */
+    xTimeY.tv_sec = 1;
+    xTimeY.tv_nsec = 2;
+    TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecCompare( &xTimeX, &xTimeY ) );
+    xTimeX.tv_sec = 0;
+    TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecCompare( &xTimeX, &xTimeY ) );
+
+}
+
+/*-----------------------------------------------------------*/
+
+TEST(  Full_POSIX_UTILS, UTILS_TimespecToTicks )
+{
+    struct timespec xTimeX = { .tv_sec = 0, .tv_nsec = 0 };
+    TickType_t xResult;
+
+    /* Check return value with NULL parameters. */
+    TEST_ASSERT_EQUAL_INT( EINVAL, UTILS_TimespecToTicks( NULL, &xResult ) );
+    TEST_ASSERT_EQUAL_INT( EINVAL, UTILS_TimespecToTicks( &xTimeX, NULL ) );
+
+    /* Check return value with invalid parameters */
+    xTimeX.tv_nsec = NANOSECONDS_PER_SECOND + 1;
+    TEST_ASSERT_EQUAL_INT( EINVAL, UTILS_TimespecToTicks( &xTimeX, &xResult ) );
+    xTimeX.tv_nsec = -1;
+    TEST_ASSERT_EQUAL_INT( EINVAL, UTILS_TimespecToTicks( &xTimeX, &xResult ) );
+
+    /* Check return value and results with valid parameters*/
+    xTimeX.tv_sec = 1;
+    xTimeX.tv_nsec = NANOSECONDS_PER_TICK + 1;
+    TickType_t xExpectedVal = xTimeX.tv_sec * configTICK_RATE_HZ + 2;
+    TEST_ASSERT_EQUAL_INT( 0, UTILS_TimespecToTicks( &xTimeX, &xResult ) );
+    TEST_ASSERT_EQUAL_INT( ( int ) xResult, ( int ) xExpectedVal);
```

Comment:
Nitpick: Expect val is on the left.

Reply:
Good Point. Will make change in next commit